### ISSUE TITLE
Assigning to `process.env` does not overwrite value

### ISSUE THREAD

**Issue by michaelhays** [oven-sh/bun#6762]
### What version of Bun is running?

1.0.7+b0393fba6200d8573f3433fb0af258a0e33ac157

### What platform is your computer?

Linux 6.2.0-35-generic x86_64 x86_64

### What steps can reproduce the bug?

Create a `.env` file:

```
FOO=first
```

Create a `run.js` file:

```js
console.log(process.env.FOO)
process.env.FOO = 'second'
console.log(process.env.FOO)
```

### What is the expected behavior?

`bun run.js` should output:

```
first
second
```

### What do you see instead?

`bun run.js` outputs:

```
first
first
```

### Additional information

Testing with Node.js 20.9.0, running `node --env-file=.env run.js` has the correct output:

```
first
second
```

**Comment by Hanaasagi** [comment]
And, delete env also doesn't work as expected.
```JavaScript
console.log(process.env.FOO);
delete process.env.FOO;
console.log(process.env.FOO);
```

This should be `undefined` after `delete`.

link https://github.com/oven-sh/bun/issues/4383

**Comment by liz3** [comment]
i would look at this tmrw unless someone does it before me just lmk here so we dont end up doing duplicated work 😅

**Comment by Hanaasagi** [comment]
Please go ahead with it; I haven't started this. @liz3 :tada:

**Comment by simylein** [comment]
This was working at some point, seems to be a regression. #2696

**Comment by michaelhays** [comment]
Fixed in Bun 1.0.15
